bag.go

```diff
@@ -108,33 +191,33 @@ func gatherErrors(errs []error, bagLocation string) string {
 	return errorMsgs
 }
 
-func CreateBag(inputDir string, algorithm string, numProcesses int) error {
+func CreateBag(inputDir string, algorithm string, numProcesses int) (Bag, error) {
 	//check that input exists and is a directory
 	if err := directoryExists(inputDir); err != nil {
```

Comment:
Would it be helpful to instantiate a `Bag` variable at the beginning of this function, e.g., `bag := Bag{}`, and then in your error branches, you could just use
`return bag, err` 
instead of
`return Bag{}, err`
?

e.g., 
```
if err != nil {
	return bag, err
}
```